packages/create-react-app/index.js

```diff
@@ -264,6 +270,30 @@ function checkAppName(appName) {
   }
 }
 
+function moveReactScriptsToDev() {
+  var packagePath = path.join(process.cwd(), 'package.json');
+  if (!pathExists(packagePath)) {
+    return;
+  }
+
+  var packageJson = require(packagePath);
+  if (typeof packageJson.dependencies === 'undefined') {
+    return;
+  }
+
+  var packageVersion = packageJson.dependencies['react-scripts']
```

Comment:
I just noticed that we already extract the package name from the `--scripts-version` even if it's a tarball URL: https://github.com/facebookincubator/create-react-app/blob/ab3913cc113b2392fa0329f699571bae3c213308/packages/create-react-app/index.js#L171

I think we can use that name to find the correct dependency from package.json.